src/tools/business/gcal_tool.py

```
@@ -128,6 +138,51 @@ def _get_calendar_tz_name(self, config: Dict[str, Any]) -> str:
+                "calendar_id": config.get("calendar_id", ""),
+                "timezone": config.get("timezone", ""),
+            }
+        return cals
+
+    def _selected_calendar_keys(self, config: Dict[str, Any]) -> list[str]:
+        sel = []
+        # Per-context nested override can set tools.google_calendar.selected_calendars
+        # When missing, default to all configured calendars
+        try:
+            sel = list(config.get("selected_calendars", []) or [])
+        except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)